### ISSUE TITLE
Automatically create ASGs

### ISSUE THREAD

**Issue by mwos-sl** [jenkinsci/ec2-fleet-plugin#306]
When I want to introduce new agent group with some labels (a.k.a. cloud fleet) I need to create ASG on AWS side.
Of course I've got it already extracted to a tf module, but still I need to add entry each time.

It would be awesome, if plugin could create it by itself. Something similar to what is already available in this plugin, but not exactly this, because the existing feature requires configuring by labels which is pretty limited.

I mean, I still would like to have configuration of the fleet in jenkins.yaml, but to have exposed some extra parameters of ASG per fleet, e.g.:
- launch_template_id
- subnet_ids
- instances for mixed_instances_policy (so that I could use several ec2 families, etc..)
- tags
- maybe something else to expose asg api as needed (some stwiches on-demand vs spot). 

I think it's fine to require creating launch_template though, because most probably it is single same one for all ec2 jenkins clouds.

Basically this is what EC2 plugin does - it does not require touching aws infra while introducing new agent type. And I think this is one of the two last reasons I still keep both "EC2 plugin" and "EC2 fleet plugin". I just sometimes need to add agent quickly and don't want to do a whole infra setup in terraform for it.

**Comment by haugenj** [comment]
Can you explain the differences between this and the [label based fleet](https://github.com/jenkinsci/ec2-fleet-plugin/blob/master/docs/LABEL-BASED-CONFIGURATION.md)?

**Comment by mwos-sl** [comment]
The difference is specifying all of the parameters via a long single string in the pipeline vs. having a single label and specifying config in yaml.

But generally I could leave with _label based fleet_ if only it supports configuring few extra [parameters](https://github.com/jenkinsci/ec2-fleet-plugin/blob/master/docs/LABEL-BASED-CONFIGURATION.md#supported-parameters) that it's missing:
1. I can't specify several instance types to increase number of available spot pools, it seems I can only provide a single instance type.
2. I'm not sure how the _label based fleet_ figure outs in which subnets it should spawn instances?
3. I can't add my custom aws tags to created resources (and I need this for cost tracking)
4. I can't add my custom init script nor `prefixStartSlaveCmd` as in yaml-based version, so that I could set up some env variables which I use per worker type, and those can't be baked into AMI. This one will be tricky to setup using single comma separaterd key-value parameters, as proposed in _label based fleet_

So TLDR: some parameters are missing and what's more important - the format of a single string with key-value comma-separated parameters in _label based fleet_ seems to me not strong enough to express more slightly more complicated configuration.

That's why I propose to let the config to be kept in .yaml.

**Comment by mwos-sl** [comment]
There is actually one more thing: I share my workers between several separate pipelines. 
I think this is not possible with _label based fleet_, right?

**Comment by mwos-sl** [comment]
@haugenj can we reopen this one? This feature is still missing and it would help vastly with creating new agents types.
